src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/buildTransitive/net6.0/Microsoft.Extensions.Telemetry.Abstractions.targets

```diff
@@ -1,11 +1,11 @@
 <Project>
   <!-- This package should replace the Microsoft.Extensions.Logging.Abstractions source generator. -->
-  <Target Name="_Microsoft_Extensions_Logging_AbstractionsRemoveAnalyzers" 
+  <Target Name="_Microsoft_Extensions_Logging_AbstractionsRemoveAnalyzers"
           Condition="'$(DisableMicrosoftExtensionsLoggingSourceGenerator)' == 'true'"
-          AfterTargets="ResolveReferences">
+          BeforeTargets="CoreCompile">
```

Comment:
I haven't followed all of the thread here, but can you explain why this needs to happen before corecompile as opposed to after RAR? Are we expecting any other target to add the analyzer that is not RAR? For things like this, having a comment explaining why we want to run at a specific time is helpful.

Reply:
You can check Igor's message: https://github.com/dotnet/extensions/issues/4969#issuecomment-2116528521
TL;DR: this works universally for both regular projects (e.g. console, web API) and WPF apps.

Reply:
In that case, can we add a comment above explaining this? While we don't have a comment there today explaining why it was after ResolveReferences, that one feels obvious as it is the one that typically add's stuff the Analyzer item. On the other hand, it is not immediately obvious why we'd put it in the before CoreCompile, and especially now that we know there is an edge case we are covering for with WPF, so I'd probably write down a comment explaining that just doing it after RAR is not enough, and share this scenario.

Also, not sure if you tested, but it would be great to validate that this also works for design time builds. (Design time builds don't run all the exact same targets that are run during a regular build, so if you don't hook properly you could end up breaking intellisense. ResolveReferences is one of the ones that does run in both regular builds and design time builds)

Reply:
`ResolveReferences` target doesn't exist in the WPF's temp project, so the removal of the analyzers
![image](https://github.com/dotnet/extensions/assets/4403806/14ab19c1-db3c-4ff8-9b48-e0efff955b47)

Intead the temp project has `RemoveDuplicateAnalyzers` target, which is run before `CoreCompile` target:
![image](https://github.com/dotnet/extensions/assets/4403806/1cf5a906-8351-4c11-94fd-d3367d92cf22)


A WPF fix would be to run our target _after_ `RemoveDuplicateAnalyzers` target, however, since that target is Windows Desktop specific, a more generic fix is to run our target before `CoreCompile`.

Reply:
@joperezr I validated that it works both in compile and design time. Both WPF and WinForms apps are working just fine.
The new [WinUI](https://learn.microsoft.com/en-us/windows/apps/winui/) doesn't work though, so I guess I will have to update the targets again.

Reply:
> The new [WinUI](https://learn.microsoft.com/en-us/windows/apps/winui/) doesn't work though, so I guess I will have to update the targets again.

@xakep139 could you share more information? E.g., a binlog.

Reply:
So, for WinUI applications the `CoreCompile` target happens after the `XamlPreCompile` target which causes the error:
![image](https://github.com/dotnet/extensions/assets/6381023/bf386f52-31a2-4c15-b1cd-d5da7e081abb)

I would recommend to stick with the initial solution and have the "standard" `BeforeTargets="ResolveReferences"` for non-WPF projects, and a separate `RemoveDuplicateAnalyzers` target for WPF ones.

Reply:
Please review the changes, folks